### PR TITLE
Add optional normal vector for customized dragging

### PR DESCRIPTION
**Pull request by nelsondude**

### CODE REVIEW

**Comment by Mugen87** [comment]
Can you please explain your use case a bit?

**Comment by nelsondude** [comment]
Currently, drag_controls assumes that when we want to drag objects in the canvas, we want to drag them normal to the camera. In the case where you want to make a 3D board game for example, and want to drag them on the plane of the board, this would be very useful. It removes dependency on the camera when dragging.

**Comment by nelsondude** [comment]
I've implemented this functionality in a 3D chess app I made.

**Review by Mugen87 — CHANGES_REQUESTED** [review]

File: examples/js/controls/DragControls.js [hunk 1]
```diff
@@ -21,11 +21,15 @@ THREE.DragControls = function ( _objects, _camera, _domElement ) {
...
 
 	var _selected = null, _hovered = null;
-
+	var _planeNormal = null;
 	//
 
 	var scope = this;
 
+	this.setPlaneNormal = function ( planeNormal ) {
+		_planeNormal = planeNormal;
```
Comment (WestLangley) [inline examples/js/controls/DragControls.js]: I would implement it like so:
```javascript
this.setPlaneNormal = function ( normal ) {

	_planeNormal.copy( normal );

};
```
You will have to modify the remainder of the changes, however.

Comment (nelsondude) [reply examples/js/controls/DragControls.js]: Thanks for the suggestion, still new to open source! Would you recommend amending or making a new commit?

Comment (WestLangley) [reply examples/js/controls/DragControls.js]: Not necessarily. You can modify this one if you want. It's up to you.

**Comment by nelsondude** [comment]
The requested changes have been added and are ready for review.

**Comment by WestLangley** [comment]
Did you test your code?

**Comment by nelsondude** [comment]
Just tested with most recent change. Used `normal.clone()` so that I dont get an error when copying `normal` into `_planeNormal` which is initially `null`.

**Comment by mrdoob** [comment]
I think what @WestLangley suggested was to initialise `_planeNormal` as `THREE.Vector3` instead of `null` so then you can do `_planeNormal.copy( normal )`.

File: examples/js/controls/DragControls.js [hunk 2]
```diff
@@ -89,7 +95,7 @@ THREE.DragControls = function ( _objects, _camera, _domElement ) {
 
 			var object = intersects[ 0 ].object;
 
-			_plane.setFromNormalAndCoplanarPoint( _camera.getWorldDirection( _plane.normal ), object.position );
+			_plane.setFromNormalAndCoplanarPoint( _planeNormal || _camera.getWorldDirection( _plane.normal ) , object.position );
```
Comment (nelsondude) [inline examples/js/controls/DragControls.js]: @mrdoob If I initialize `_planeNormal` as `THREE.Vector3`, how would you suggest modifying the above line?

Comment (mrdoob) [reply examples/js/controls/DragControls.js]: Oh, I see...

File: examples/js/controls/DragControls.js [hunk 3]
```diff
@@ -204,7 +210,7 @@ THREE.DragControls = function ( _objects, _camera, _domElement ) {
 
 			_selected = intersects[ 0 ].object;
 
-			_plane.setFromNormalAndCoplanarPoint( _camera.getWorldDirection( _plane.normal ), _selected.position );
+			_plane.setFromNormalAndCoplanarPoint( _planeNormal || _camera.getWorldDirection( _plane.normal ) , object.position );
```
Comment (mrdoob) [inline examples/js/controls/DragControls.js]: I think this should still use ` _selected.position`.

**Comment by Mugen87** [comment]
Closing since the PR is incomplete and has merge conflicts.

@mrdoob If you are still interested in this feature, I can make a new PR with all relevant changes (unless @nelsondude wants to do this^^).

**Pull request closed without merging**